File: frontend/src/types/billing.ts
import { PremiumFrequency } from './product';

export interface PremiumInstallment {
  id: string;
  policyId: string;
  policyNumber: string;
  productName: string;
  dueDate: string;
  amount: number;
  status: 'PENDING' | 'PAID' | 'OVERDUE' | 'CANCELLED';
  paymentDate?: string;
  transactionId?: string;
  receiptNumber?: string;
}

export interface PaymentReceipt {
  id: string;
  receiptNumber: string;
  policyNumber: string;
  productName: string;
  amount: number;
  paymentDate: string;
  paymentMethod: string;
  downloadUrl: string;
}

export interface BillingDashboardVM {
  summary: {
    totalPaid: number;
    totalOutstanding: number;
    nextPaymentDate: string;
    nextPaymentAmount: number;
  };
  recentTransactions: PremiumInstallment[];
}

File: backend/app/schemas/billing.py
from app.schemas.base import BaseSchema, AuditSchema
from datetime import date
from typing import List, Optional
from uuid import UUID

class InstallmentRead(AuditSchema):
    policy_id: UUID
    amount: float
    due_date: date
    status: str
    payment_date: Optional[date] = None
    transaction_id: Optional[str] = None
    receipt_number: Optional[str] = None

class ReceiptRead(AuditSchema):
    receipt_number: str
    amount: float
    payment_date: date
    payment_method: str
    download_url: str

class BillingDashboardRead(BaseSchema):
    total_paid: float
    total_outstanding: float
    next_payment_date: Optional[date]
    next_payment_amount: float
    recent_transactions: List[InstallmentRead]
